Fig.Core/Settings.cs

```diff
@@ -154,45 +177,117 @@ public T Bind<T>(bool requireAll = true, string prefix = null) where T : new()
         /// <typeparam name="T"></typeparam>
         public void Bind<T>(T target, bool requireAll = true, string prefix = null) where T : new()
         {
-            prefix = ((prefix ?? String.Empty).Trim() == String.Empty)
+            var errors = new List<string>();
+
+            var result = GetBindResult(target, requireAll, prefix, false, errors);
+
+            if (result.Errors.Any()) throw new ConfigurationException(errors);
+        }
+
+        private BindResult<T> GetBindResult<T>(bool requireAll = true, string prefix = null, bool preload = false, List<string> errors = null) where T : new()
+        {
+            var t = new T();
+            return GetBindResult(t, requireAll, prefix, preload, errors);
+        }
+
+        private BindResult<T> GetBindResult<T>(T target, bool requireAll = true, string prefix = null, bool preload = false, List<string> errors = null) where T : new()
+        {
+            errors = errors ?? new List<string>();
+
+            prefix = ((prefix ?? String.Empty).Trim() == String.Empty && !preload)
                 ? typeof(T).Name
                 : prefix;
 
-            var props = typeof(T)
-                .GetProperties()
-                .Where(p => !(p.GetSetMethod() is null));
-
-            foreach (var prop in props)
+            foreach (var prop in typeof(T).GetProperties())
             {
-                var type = prop.PropertyType;
-                var name = $"{prefix}.{prop.Name}";
-                object value = null;
-
-                if (!type.IsPrimitive
-                    && !type.IsEnum
-                    && type != typeof(string)
-                    && type != typeof(decimal)
-                    && type != typeof(DateTime))
+                try
                 {
-                    value = this.GetType().GetMethods()
-                        .Where(x => x.Name == "Bind" && x.GetParameters()?.Length == 2)
-                        .FirstOrDefault()?
-                        .MakeGenericMethod(type)?
-                        .Invoke(this, new object[] { requireAll, name });
+                    if (preload && prop.Name == nameof(Environment)) continue;
+
+                    var readonlyProp = prop.GetSetMethod() is null;
+                    var type = prop.PropertyType;
+                    var name = String.IsNullOrEmpty(prefix?.Trim()) ? prop.Name : $"{prefix}.{prop.Name}";
+                    object value = null;
+
+                    if (preload || !readonlyProp)
+                    {
+                        if (!type.IsPrimitive
+                            && !type.IsEnum
+                            && type != typeof(string)
+                            && type != typeof(decimal)
+                            && type != typeof(DateTime)
+                            && type != typeof(TimeSpan))
```

Comment:
Extract to method `IsNestedType()`  perhaps? Also put the types in a HashSet<Type> and do a Contains lookup